Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -295,17 +358,50 @@ void __stdcall PlayerLaunch(unsigned int iShip, unsigned int client)
 	}
 }
 
+bool __stdcall LaunchPosHook(uint space_obj, struct CEqObj &p1, Vector &pos, Matrix &rot, int dock_mode)
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	// Redirect the ship to carrier's position. Can bug with POB plugin, changes may be required.
+	if (undockingShip.proxyBaseID == space_obj)
+	{
+		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
+
+		rot = undockingShip.carrierRot;
+		pos = undockingShip.carrierPos;
+		undockingShip.proxyBaseID = 0;
+	}
+	return true;
+}
+
 void __stdcall PlayerLaunch_AFTER(unsigned int ship, unsigned int client)
 {
 	returncode = DEFAULT_RETURNCODE;
 
 	// Is the client flagged to dock at a base after exiting?
-	if(mobiledockClients[client].baseUndock)
+	if (mobiledockClients[client].baseUndock)
 	{
 		pub::Player::ForceLand(client, mobiledockClients[client].undockBase->iBaseID);
 		mobiledockClients.erase(client);
 	}
 
+	// Land ship to proxy base in carrier's system if they are in different systems.
+	if (mobiledockClients[client].carrierSystem == -1)
+	{
+		if (!mobiledockClients[client].wscDockedWithCharname.empty())
+		{
+			string scProxyBase = HkGetPlayerSystemS(HkGetClientIdFromCharname(mobiledockClients[client].wscDockedWithCharname)) + "_proxy_base";
+			if (pub::GetBaseID(mobiledockClients[client].proxyBaseID, scProxyBase.c_str()) == -4)
+			{
+				PrintUserCmdText(client, L"No proxy base in system detected. Contact a developer about this please. Required base: %s", scProxyBase);
+				return;
+			}
+		}
+
+		pub::Player::ForceLand(client, mobiledockClients[client].proxyBaseID);
+		PrintUserCmdText(client, L"Navmap updated successfully.");

```

Comment:
this message makes no sense?

Reply:
Makes with added commentary.

Reply:
Invoker PM'd me: 'When carrier goes to another system, docked ships remain in previous with outdated system navigational map.'
Resolving